core/model/library.py

```diff
@@ -189,8 +192,25 @@ class Library(Base, HasSessionCache):
         uselist=False,
     )
 
-    # Typing specific
-    collections: List[Collection]
+    @property
+    def collections(self) -> Sequence[Collection]:
+        """Get the collections for this library"""
+        from core.model import (
+            Collection,
+            IntegrationConfiguration,
+            IntegrationLibraryConfiguration,
+        )
+
+        _db = Session.object_session(self)
+        return _db.scalars(
+            select(Collection)
+            .join(IntegrationConfiguration)
+            .join(IntegrationLibraryConfiguration)
+            .where(
+                IntegrationConfiguration.goal == Goals.LICENSE_GOAL,
+                IntegrationLibraryConfiguration.library_id == self.id,
+            )
+        ).all()
```

Comment:
Turning this into a read only property caused a lot of little changes in tests, not much in our production code though. Instead of doing:
`library.collections.append(collection)` we end up doing `collection.libraries.append(library)`